app/lib/dfe/bigquery/test_helper.rb

```diff
@@ -7,6 +7,7 @@ def stub_bigquery_application_metrics_request
       end
 
       def stub_bigquery_client
+        @bigquery_client = nil
```

Comment:
Why do you need this?

Reply:
But you are setting it to `nil` before calling `bigquery_client` so at the end of the `stub_bigquery_client` method call wouldn't it always have the same value of `InstanceDouble(Google::Cloud::Bigquery::Project)`?

Reply:
> But you are setting it to `nil` before calling `bigquery_client` so at the end of the `stub_bigquery_client` method call wouldn't it always have the same value of `InstanceDouble(Google::Cloud::Bigquery::Project)`?

Yes, but not the same object stub - object id in memory (it needs to be a different stub for every spec otherwise RSpec will complain about leaking the stub to other specs).

Reply:
This doesn't seem right. I'm still trying to figure out what you're trying to accomplish here. It might become clearer in future PRs.

Why do you need those different stubs? Why not:

```ruby
def stub_bigquery_application_metrics_request
  bigquery_client = instance_double(Google::Cloud::Bigquery::Project)
  allow(DfE::Bigquery).to receive(:client).and_return(bigquery_client)
  allow(bigquery_client).to receive(:query).and_return(application_metrics_results)
end
```

Reply:
Your suggestion is much better. I added that one above. I think I am working almost 2 weeks on this report and I am losing the big perspective and for some reason trying to over complicate things. (today morning I realise when I simplified the CSV grouping of the data). Either way I did what suggested above.